src/js/app.js

```diff
@@ -622,6 +780,170 @@ document.addEventListener("DOMContentLoaded", function () {
+  if (storageForm) {
+    storageForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+
+      if (!storageService) {
+        showToast('Storage service not initialized. Please provide a storageBucket in configuration.');
+        output(`<b>Error:</b> Storage service not available. Please reinitialize with a storageBucket.`);
+        return;
+      }
+
+      let storage_path = storageForm["storage_path"].value;
+      let storage_op = storageForm["storage_op"].value;
+      let storage_limit = parseInt(storageForm["storage_limit"].value) || 100;
+      let file_upload = storageForm["file_upload"].files[0];
+
+      switch (storage_op) {
+        case 'list':
+          try {
+            let listRef = storage_path ? storageService.ref(storage_path) : storageService.ref();
+            listRef.listAll().then(result => {
+              let itemsCount = Math.min(result.items.length, storage_limit);
+              let foldersCount = Math.min(result.prefixes.length, storage_limit);
+              
+              let output_text = `<b>Listing storage contents</b><br/>`;
+              output_text += `<b>Path:</b> ${storage_path || '(root)'}<br/>`;
+              output_text += `<b>Files (${itemsCount}/${result.items.length}):</b><br/>`;
+              
+              for (let i = 0; i < itemsCount; i++) {
+                let item = result.items[i];
+                output_text += `📄 ${item.fullPath}<br/>`;
+              }
+              
+              output_text += `<b>Folders (${foldersCount}/${result.prefixes.length}):</b><br/>`;
+              for (let i = 0; i < foldersCount; i++) {
+                let prefix = result.prefixes[i];
+                output_text += `📁 ${prefix.fullPath}<br/>`;
+              }
+              
+              output(output_text);
+            }).catch(error => {
+              output(`<b>Error listing storage:</b> ${error.message}`);
+              console.error('Storage list error:', error);
+            });
+          } catch (error) {
+            output(`<b>Error:</b> ${error.message}`);
+            console.error('Storage list error:', error);
+          }
+          break;
+
+        case 'upload':
+          if (!file_upload) {
+            showToast('Please select a file to upload');
+            return;
+          }
+          
+          if (!storage_path) {
+            showToast('Please specify a storage path for the upload');
+            return;
+          }
+
+          try {
+            let uploadRef = storageService.ref(storage_path);
+            let uploadTask = uploadRef.put(file_upload);
+            
+            output(`<b>Uploading file:</b> ${file_upload.name} to ${storage_path}...`);
+            
+            uploadTask.on('state_changed', 
+              (snapshot) => {
+                let progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+                output(`<b>Upload progress:</b> ${progress.toFixed(1)}%`);
```

Comment:
nice, but `snapshot.totalBytes` can be 0 (empty hello.txt file), which creates division by zero(`progress: NaN%`).
let's avoid printing the value of `progress` if it's NaN